src/modules/twap/hooks/useQuoteObserver.ts

```diff
@@ -0,0 +1,27 @@
+import { useMemo } from 'react'
```

Comment:
This hook will observe the quote changes and update the output amount when necessary.